metrics/src/main/java/org/jboss/pnc/metrics/MetricsConfiguration.java

```diff
@@ -0,0 +1,163 @@
+
+        String value;
+
+        String valueSys = System.getProperty(propertyName);
+        String valueEnv = System.getenv(propertyName);
+
+        if (valueSys != null) {
+            value = valueSys;
+        } else if (valueEnv != null) {
+            value = valueEnv;
+        } else {
+            throw new NoPropertyException("Property '" + propertyName + "' not specified");
+        }
+
+        logger.info("Updated " + description + " to: " + value);
+
+        return value;
+    }
+
+
+    /**
+     * Add metrics for JVM. Already provided by metrics-jvm
+     */
+    private void monitorJvmMetrics() {
+
+        logger.info("Registering JVM metrics");
+
+        metricRegistry.register(METRIC_JVM_GARBAGE, new GarbageCollectorMetricSet());
+        metricRegistry.register(METRIC_JVM_MEMORY, new MemoryUsageGaugeSet());
+        metricRegistry.register(METRIC_JVM_THREADS, new ThreadStatesGaugeSet());
+        metricRegistry.register(METRIC_JVM_CLASSLOADING, new ClassLoadingGaugeSet());
+
+    }
+
+    /**
+     * Read system or environment variable to know to which Graphite server to report data to.
+     *
+     * If any of the required variables are not specified, abandon the setup and warn the user.
+     *
+     */
+    private void setupGraphiteReporter() {
+        try {
+
+            String graphiteServer = getValueFromProperty(GRAPHITE_SERVER_KEY, "Graphite Server URL");
+            int graphitePort = Integer.parseInt(getValueFromProperty(GRAPHITE_PORT_KEY, "Graphite Port"));
+            String graphitePrefix = getValueFromProperty(GRAPHITE_PREFIX_KEY, "Graphite Prefix");
+
+            startGraphiteReporter(graphiteServer, graphitePort, graphitePrefix);
+
+        } catch (NumberFormatException e) {
+
+            // thrown because it couldn't parse graphitePort
+            logger.warn("Could not parse Graphite port! Aborting reporting data to Graphite", e);
+
+        } catch (NoPropertyException e) {
+
+            logger.warn("Could not find property required to setup Graphite reporting! Reason: {}", e.getMessage());
+
+        }
+    }
+
+    /**
+     * Reporter of metrics to a Graphite server
+     *
+     * @param host   Graphite server
+     * @param port   Graphite port (usually 2003)
+     * @param prefix Prefix to use as a namespace for our logs
+     */
+    private void startGraphiteReporter(String host, int port, String prefix) {
+
+        logger.info("Setting up Graphite reporter");
+
+        Graphite graphite = new Graphite(new InetSocketAddress(host, port));
+
+        GraphiteReporter reporter = GraphiteReporter.forRegistry(metricRegistry)
+                .prefixedWith(prefix)
+                .convertRatesTo(TimeUnit.SECONDS)
+                .convertDurationsTo(TimeUnit.MILLISECONDS)
+                .filter(MetricFilter.ALL)
+                .build(graphite);
+
+        reporter.start(1, TimeUnit.MINUTES);
```

Comment:
I would be useful to have a configurable metrics interval. Maybe in seconds.

Reply:
I will add it in my next PR for v1.3 and master!